### DISCUSSION TITLE
Use SnapKit for Wallet app

### DISCUSSION THREAD

**Discussion by anhnguyenbitmark** [BlockchainCommons/Gordian#77] [General Discussion]
Hi @wolfmcnally @Fonta1n3 ,
Would you recommend using [SnapKit](https://github.com/SnapKit/SnapKit) for iOS wallet apps?

As SnapKit defines DSL to make Auto Layout easy, it will save us a bit effort in making UI of the app. And since we anticipate that the UI will be frequently changed during the development, having SnapKit's help would be a significant advantage.

But we are still worry for its security factor. That's why we would love to have your advices for this.

**Comment by Fonta1n3** [comment]
Hi @anhnguyenbitmark,

I’ve never used SnapKit and can’t comment on its security.

I generally avoid third party libraries unless they are absolutely required but that’s just my personal preference and not necessarily because of security concerns.

**Reply by anhnguyenbitmark** [reply]
Thanks for your reply.

I got the idea here.
It's not absolutely required for us to have a UI library. But it can help to speed up the development.
We will carefully consider the trade-off to use it.

**Comment by wolfmcnally** [comment]
I don't know anything about SnapKit, so I can't recommend it. If you can use SwiftUI, that is what I recommend. SwiftUI includes its own layout DSL directly in Swift.

**Reply by anhnguyenbitmark** [reply]
Thanks for your reply.

Certainly we love to give a try with SwiftUI. But our app target is iOS 13, SwiftUI in that version is not mature enough. So we'd like to take a safe approach with UIKit first, and look forward for a migration later on. That's also a factor that we wanted to bring SnapKit in, as it also has layout DSL, not similar but might be easier to migrate to SwiftUI later.
